aries/agents/rust/mediator/mediation/src/routes/forward.rs

```diff
@@ -5,20 +5,50 @@ use std::sync::Arc;
 
 use axum::{extract::State, Json};
 use log::{debug, info};
+use messages::{
+    decorators::thread::Thread,
+    msg_fields::protocols::{
+        notification::ack::{Ack, AckContent, AckDecorators, AckStatus},
+        routing::Forward,
+    },
+};
+use uuid::Uuid;
 
-use crate::{didcomm_types::ForwardMsg, storage::MediatorPersistence};
+use crate::storage::MediatorPersistence;
 
 pub async fn handle_forward<T>(
     State(storage): State<Arc<T>>,
-    Json(forward_msg): Json<ForwardMsg>,
-) -> Json<ForwardMsg>
+    Json(forward_msg): Json<Forward>,
+) -> Json<Ack>
 where
     T: MediatorPersistence,
 {
     info!("Persisting forward message");
     debug!("{forward_msg:#?}");
-    let _ = storage
-        .persist_forward_message(&forward_msg.recipient_key, &forward_msg.message_data)
-        .await;
-    Json(forward_msg)
+    let _ack_status = match storage
+        .persist_forward_message(
+            &forward_msg.content.to,
+            &serde_json::to_string(&forward_msg.content.msg).unwrap(),
+        )
+        .await
+    {
+        Ok(_) => {
+            info!("Persisted forward");
+            AckStatus::Ok
+        }
```

Comment:
In next PR all these functions will return Result<T>. 
Error will be bubbled up for processing. Which can then be converted into `ProblemReport` message.

 To keep consistent return signature here, using Ack status Pending enum, as indication that something is not right.